circom/circom_test.go

```diff
@@ -0,0 +1,527 @@
+package circom
+
+import (
+	"fmt"
+	"github.com/consensys/gnark-crypto/ecc"
+	fr_bls12381 "github.com/consensys/gnark-crypto/ecc/bls12-377/fr"
+	bls12381 "github.com/consensys/gnark-crypto/ecc/bls12-381"
+	fr_bn254 "github.com/consensys/gnark-crypto/ecc/bn254/fr"
+	"github.com/consensys/gnark-crypto/ecc/secp256k1"
+	"github.com/consensys/gnark-crypto/ecc/secp256k1/fp"
+	fr_secp "github.com/consensys/gnark-crypto/ecc/secp256k1/fr"
+	groth16 "github.com/consensys/gnark/backend/groth16/bn254"
+	"github.com/consensys/gnark/backend/groth16/bn254/mpcsetup"
+	"github.com/consensys/gnark/constraint"
+	cs "github.com/consensys/gnark/constraint/bn254"
+	"github.com/consensys/gnark/frontend"
+	"github.com/consensys/gnark/frontend/cs/r1cs"
+	"github.com/consensys/gnark/std/algebra/emulated/sw_emulated"
+	zksha3 "github.com/consensys/gnark/std/hash/sha3"
+	"github.com/consensys/gnark/std/math/bits"
+	"github.com/consensys/gnark/std/math/emulated"
+	"github.com/consensys/gnark/std/math/uints"
+	"github.com/consensys/gnark/test"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/crypto/ecies"
+	"golang.org/x/crypto/sha3"
+	"math/big"
+	"math/rand"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+)
+
+func doMPCSetUp(ccs constraint.ConstraintSystem) (pk groth16.ProvingKey, vk groth16.VerifyingKey, err error) {
+	const (
+		nContributionsPhase1 = 3
+		nContributionsPhase2 = 3
+		power                = 21 //element count range 2^0-2^27
+	)
+
+	initPhase1 := mpcsetup.InitPhase1(power)
+
+	FilePhase1Init, err := os.Create("Phase1_1")
```

Comment:
defer os.Remove("Phase1_*")
defer os.Remove("Phase2_*")

Reply:
It’s not good to write like this. These are not temporary files and cannot be deleted